api/grpcserver/mesh_service.go

```diff
@@ -352,8 +347,17 @@ func (s MeshService) readLayer(ctx context.Context, layerID types.LayerID, layer
 	// Add unique ATXIDs
 	atxs, matxs := s.mesh.GetATXs(ctx, activations)
 	if len(matxs) != 0 {
-		s.logger.With().Error("could not find activations from layer",
-			log.String("missing", fmt.Sprint(matxs)), layer.Index())
+		ctxzap.Error(
+			ctx,
+			"could not find activations from layer",
+			zap.Array("missing", zapcore.ArrayMarshalerFunc(func(ae zapcore.ArrayEncoder) error {
+				for _, atxid := range matxs {
+					ae.AppendString(atxid.ShortString())
+				}
+				return nil
```

Comment:
Since this seems to be used in multiple places it might make sense to add a general way to do this. In the file `ATXID` is defined we can add:

```golang
type ATXIDs []ATXID

func (ids ATXIDs) MarshalLogArray(enc zapcore.ArrayEncoder) error {
	for _, id := range ids {
		enc.AppendString(id.ShortString())
	}
	return nil
}
```

and then when using it:

```golang
ctxzap.Error(ctx, "could not find activations from layer",
	zap.Array("missing", types.ATXIDs(matxs)),
	layer.Index().Field().Zap(),
)
```